api/v1beta1/seedimage_type.go

```diff
@@ -45,6 +49,24 @@ type SeedImageSpec struct {
 	CloudConfig map[string]runtime.RawExtension `json:"cloud-config,omitempty" yaml:"cloud-config,omitempty"`
 }
 
+type BuildContainer struct {
```

Comment:
Rather than creating a devoted struct for that I think it would be easier to just reuse the k8s objects, this way we also can re-use it internally for creating pods when required. See the `ManagedOSImageSpec` and the ContainerSpec used there for the UpgradeContainer field.

Reply:
I actually used a `corev1.Container` in the draft of this PR but was thinking that this minimal implementation is all that is needed for now so I didn't want to expose the entire `corev1.Container` object.